src/Component/Layer/LayerTypeSelect/LayerTypeSelect.tsx

```diff
@@ -0,0 +1,70 @@
+import React, {
+  useEffect,
+  useState
+} from 'react';
+
+import {
+  Select
+} from 'antd';
+import { SelectProps } from 'antd/lib/select';
+
+export enum LayerType {
+  TILEWMS = 'TILEWMS',
+  VECTORTILE = 'VECTORTILE',
+  WFS = 'WFS',
+  WMS = 'WMS',
+  WMTS = 'WMTS',
+  XYZ = 'XYZ',
+  WMSTime = 'WMSTime',
+}
+
+type SelectPropsExcludes = 'defaultValue' | 'value' | 'options' | 'onChange';
+
+export interface LayerTypeSelectProps extends Omit<SelectProps<LayerType>, SelectPropsExcludes> {
+  value?: LayerType;
+  onChange?: (value: string) => void;
+};
+
+export const LayerTypeSelect: React.FC<LayerTypeSelectProps> = ({
+  value,
+  onChange,
+  ...passThroughProps
+}) => {
+
+  const [currentValue, setCurrentValue] = useState<LayerType>(value);
+
+  useEffect(() => {
+    setCurrentValue(value);
+  }, [value]);
+
+  const getOptions = (): any[] => {
+    const options = [];
+
+    for (const value in LayerType) {
+      options.push({
+        label: value,
+        value: value
+      });
+    }
+
+    return options;
+  };
+
+  const changeHandler = (value: LayerType) => {
+    setCurrentValue(value);
+
+    onChange(value);
+  };
+
+  return (
+    <Select
+      defaultValue={currentValue}
+      value={currentValue}
+      options={getOptions()}
```

Comment:
Minor: Calling a function here is not needed when using a FC. You could just use the options array and omit the function.

Reply:
True, but the `options` must be created/filled beforehand and I like to have this encapsulated in a function.